Clarify opensearch installation requirements (#2003)
shopware/docs: guides/hosting/infrastructure/elasticsearch/elasticsearch-setup.md

## Overview

As soon as several thousand data sets are used in a project, it makes sense to deal with Elasticsearch. The Elasticsearch integration for Shopware is in the [shopware/elasticsearch](https://github.com/shopware/elasticsearch) bundle. If this is not available in your project, you can add it via `composer require shopware/elasticsearch`. In this documentation, we will provide you with a short overview of the functionalities of Elasticsearch on your server and the configuration, activation, and indexing process in Shopware for live and test environments.

::: info
Currently, the implementation for Elasticsearch/Opensearch works in the same way.
:::

## Requirements

> * A supported OpenSearch (or ElasticSearch) server

Reviewer: => "Elasticsearch"

The current Shopware 6 integration is designed to work with the out-of-the-box configuration of Elasticsearch. This does not mean, of course, that these are the best settings for a production environment. Although they will affect performance and security, the settings you choose to use on your Elasticsearch setup will be mostly transparent to your Shopware installation. The best setting constellation for your shop will greatly depend on your server setup, the number, and structure of products, and replication requirements, to name a few. In this document, we can't give you specific examples for your setup, but provide you with hints and basics you might need to choose your perfect setup. More detailed information can be found on the official [Elasticsearch](https://www.elastic.co/guide/index.html) documentation page.

### Elasticsearch server setup

Elasticsearch is meant to be used as a cluster setup so it can scale properly and provide you with reliability.
In this cluster, you can choose how many nodes you want to use and which different type each node in the cluster shall have.

> A one-node cluster is only meant for development or test environments because it can't scale at all and does not give you any more reliability.

Reviewer: ```suggestion
A one-node cluster should only be used for development or test environments, because it can't scale and does not provide additional reliability.
```

A productive cluster of Elasticsearch should always contain 3 nodes that are all master-eligible nodes set by the `node.master` property in the elasticsearch.yml file. The master node is chosen by an election process of which only the master-eligible nodes are part. In an election process, you have to mind a quorum of master-eligible nodes, so you get a specific result of the election, so you should have N/2+1 master-eligible nodes. 3 is the minimum number for this because then the currently elected master node fails, you can still have a correct election process for a new master. The setting "cluster.initial_master_nodes: ["masternode1","masternode2","masternode3"]" should be provided on each of those master-eligible nodes on start.

#### Ingest nodes

Ingest nodes provide the ability to pre-process a document before it gets indexed.
The ingest node intercepts bulk and index requests, applies transformations, and then passes the documents back to the index or bulk APIs.
All nodes are Ingest nodes by default. This can be changed by the `node.ingest` property in the elasticsearch.yml file.

#### Data nodes

> Data nodes have two main features. They hold the [shards](#shards) that contain the documents/elements you have indexed and execute data-related operations like CRUD, search, and aggregations.

Reviewer: ==> "features:" (adds colon)

You can start multiple messenger consumer processes by using the command `bin/console messenger:consume` and also add output to the processed messages by adding the parameter `bin/console messenger:consume -vv`.
In a production environment, you want to deactivate the admin messenger which is started automatically when opening a session in your Administration view by following this [documentation](/docs/guides/plugins/plugins/framework/message-queue/add-message-handler#the-admin-worker).

> Our experience has shown that up to three worker processes are normal and useful for a production environment.

Reviewer: +1 for spelling out single numbers for readability—would change other small numbers in the page similarly for consistency
